modules/calib3d/src/usac/degeneracy.cpp

```diff
@@ -0,0 +1,67 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+/*
+* Abstract class for handling degeneracies in RANSAC.
+*/
+#include "../precomp.hpp"
+#include "../usac.hpp"
+
+namespace cv { namespace usac {
+
+class HomographyDegeneracyImpl : public HomographyDegeneracy {
```

Comment:
check this https://github.com/opencv/opencv/wiki/Coding_Style_Guide#steps-to-make-your-class-following-this-style

Reply:
If I don't use public I will have error. Did you mean that?